### PR TITLE
chore: Improves testing of edge cases and error states

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

Improvements to constraints in db_users:
- Username can't be empty
- Password can't be empty
- Permission can't be other than 0 or 1

Improvements in db_certificates:
- Fixes the name of `DeleteCertificate` it was mistakenly named `DeleteCertificateRequestByCSR`
- `GetCertificateChain` expects `Certificate` objects not `CertificateRequestWithChain`
- The mapping between the values returned from the `getCertificateChainStmt` won't match the fields in the `Certificate` object if the `private_key_id` is not queried.

Improvements in db_csrs.go:
- Fixes naming mismatches in docstrings
- Fixes docstring of `DeleteCertificateRequest`

Test improvements:
- Tests edge cases and error states of `CreateUser`, `GetUser`, `UpdatePassword` and `DeleteUser` in `db_users_test`
- Test edge cases and error states of `GetCertificateChain` and improve the `TestCertificatesEndToEnd` to cover `DeleteCertificate` and `GetCertificateChain`
- Tests edge cases and error states for `DeleteCertificateRequest`, `RevokeCertificateRequest` and `RejectCertificateRequest` and adds tests for missing functions like `AddCertificateChainToCertificateRequest` and `GetCertificateRequestAndChain` in the `TestCSRsEndToEnd`. 

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]
Looks good, I do have a small comment however.

File: internal/db/db_certificates_test.go [hunk 1]
```diff
@@ -78,35 +78,84 @@ func TestCertificatesEndToEnd(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("Couldn't complete List: %s", err)
+	}
+	if len(certs) != 3 {
+		t.Fatalf("Expected 3 Certificates, only got %d", len(certs))
+	}
 }
 
-func TestCreateCertificateFails(t *testing.T) {
+func TestGetCertificateFails(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (saltiyazan) [reply internal/db/db_certificates_test.go:123]: `AddCertificateChainToCertificateRequest` is tested in the `TestCertificatesEndToEnd` and in `TestCertificateAddFails`. In `TestGetCertificateFails` it is used to set up the test where we check what happens when we try to get a certificate with non-valid filters.

[the remark this answers is not in the document]
Comment (ghislainbourgeois) [reply internal/db/db_certificates_test.go:123]: Ok

[the remark this answers is not in the document]
Comment (kayra1) [reply internal/db/db_certificates_test.go:123]: This is why I kept the nolint's here, because this function is not where we check for these errors and it becomes confusing when reading from top down. Maybe we should keep them just to avoid this confusion.

**Review by ghislainbourgeois — APPROVED** [review]

**Review by kayra1 — CHANGES_REQUESTED** [review]
couple philosophical things, thanks

File: internal/db/db_certificates_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (saltiyazan) [reply internal/db/db_certificates_test.go:123]: I see, it makes sense. (I added them back)

**Review by kayra1 — APPROVED** [review]
